app/assets/javascripts/discourse/app/lib/sidebar/user/categories-section/category-section-link.js

```diff
@@ -79,6 +80,8 @@ export default class CategorySectionLink {
   }
 
   get route() {
-    return "discovery.category";
+    return this.currentUser?.sidebarTopicDestination === "unread"
+      ? "discovery.newCategory"
+      : "discovery.latestCategory";
```

Comment:
Another note here is that we should be linking to `discovery.category` and not `discovery.latestCategory`. See https://github.com/discourse/discourse/commit/90f1e2df8020dbb552e414ed26efc687c4ab16ba for more context.